### DISCUSSION TITLE
Option to validate textual or binary Avro data against a schema without decoding it?

### DISCUSSION THREAD

**Discussion by bruth** [hamba/avro#412] [General]
Hi, I have a use case to validate encoded data against a schema, but I don't actually need to decode it in a native Go value. It appears there is not method to do this today, so if I do pass a value, what would be the most efficient way to achieve this?

**Comment by nrwiersma** [comment]
Hi. The only way to validate the payload is to read the entire payload. You can try decide to a `struct{}`. In theory this should work, but I have never tried it.
